x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderReasonWireSerializationTests.java

```diff
@@ -15,63 +19,109 @@
-                    instance.assignedShardIds()
+                    instance.assignedShardIds(),
+                    instance.unassignedAllocationResults(),
+                    instance.assignedAllocationResults()
                 );
             case 1:
                 return new ReactiveStorageDeciderService.ReactiveReason(
                     instance.summary(),
                     randomValueOtherThan(instance.unassigned(), ESTestCase::randomNonNegativeLong),
                     instance.unassignedShardIds(),
                     instance.assigned(),
-                    instance.assignedShardIds()
+                    instance.assignedShardIds(),
+                    instance.unassignedAllocationResults(),
+                    instance.assignedAllocationResults()
                 );
             case 2:
                 return new ReactiveStorageDeciderService.ReactiveReason(
                     instance.summary(),
                     instance.unassigned(),
                     instance.unassignedShardIds(),
                     randomValueOtherThan(instance.assigned(), ESTestCase::randomNonNegativeLong),
-                    instance.assignedShardIds()
+                    instance.assignedShardIds(),
+                    instance.unassignedAllocationResults(),
+                    instance.assignedAllocationResults()
                 );
             case 3:
                 return new ReactiveStorageDeciderService.ReactiveReason(
                     instance.summary(),
                     instance.unassigned(),
                     new TreeSet<>(randomUnique(() -> new ShardId(randomAlphaOfLength(8), UUIDs.randomBase64UUID(), randomInt(5)), 8)),
                     instance.assigned(),
-                    instance.assignedShardIds()
+                    instance.assignedShardIds(),
+                    instance.unassignedAllocationResults(),
+                    instance.assignedAllocationResults()
                 );
             case 4:
                 return new ReactiveStorageDeciderService.ReactiveReason(
                     instance.summary(),
                     instance.unassigned(),
                     instance.unassignedShardIds(),
                     instance.assigned(),
-                    new TreeSet<>(randomUnique(() -> new ShardId(randomAlphaOfLength(8), UUIDs.randomBase64UUID(), randomInt(5)), 8))
+                    new TreeSet<>(randomUnique(() -> new ShardId(randomAlphaOfLength(8), UUIDs.randomBase64UUID(), randomInt(5)), 8)),
+                    instance.unassignedAllocationResults(),
+                    instance.assignedAllocationResults()
                 );
             case 5:
                 return new ReactiveStorageDeciderService.ReactiveReason(
                     instance.summary(),
                     instance.unassigned(),
                     new TreeSet<>(randomUnique(() -> new ShardId(randomAlphaOfLength(8), UUIDs.randomBase64UUID(), randomInt(5)), 8)),
```

Comment:
I think this should be `instance.unassignedShardIds()`?

Reply:
I think it's needed for the case where we have both random unassigned and assigned shard ids